compiler/rustc_ast_lowering/src/item.rs

```diff
@@ -1376,7 +1376,6 @@ impl<'hir> LoweringContext<'_, 'hir> {
             self.lifetime_res_to_generic_param(ident, node_id, res)
         }));
 
-        let has_where_clause_predicates = !generics.where_clause.predicates.is_empty();
```

Comment:
Checking that the AST generics' predicates is empty is not the same as checking that the HIR generics' prredicates is empty, since we process some bounds like `Sized`.